src/extension.ts

```diff
@@ -61,18 +65,32 @@ export function activate(context: vscode.ExtensionContext) {
 	initAsyncCommand(context, 'cosmosDB.deleteMongoDB', (element: MongoDatabaseNode) => deleteDatabase(element));
 	initAsyncCommand(context, 'cosmosDB.deleteDocDBDatabase', (element: DocDBDatabaseNode) => CosmosDBCommands.deleteDocDBDatabase(element, explorer));
 	initAsyncCommand(context, 'cosmosDB.deleteDocDBCollection', (element: DocDBCollectionNode) => CosmosDBCommands.deleteDocDBCollection(element, explorer));
+	initAsyncCommand(context, 'cosmosDB.deleteDocDBDocument', (element: DocDBDocumentNode) => CosmosDBCommands.deleteDocDBDocument(element, explorer));
 	initCommand(context, 'cosmosDB.newMongoScrapbook', () => createScrapbook());
 	initCommand(context, 'cosmosDB.executeMongoCommand', () => lastCommand = MongoCommands.executeCommandFromActiveEditor(connectedDb));
-	initCommand(context, 'cosmosDB.updateMongoDocuments', () => MongoCommands.updateDocuments(connectedDb, lastCommand));
+	initCommand(context, 'cosmosDB.updateDocuments', () => {
+		if (lastOpenedDocumentType === "Mongo") {
+			MongoCommands.updateDocuments(connectedDb, lastCommand);
+		}
+		else if (lastOpenedDocumentType === "DocDB") {
+			CosmosDBCommands.updateDocDBDocument(lastOpenedDocument);
+		}
+	}
+	);
 	initCommand(context, 'cosmosDB.openMongoCollection', (collection: MongoCollectionNode) => {
 		connectToDatabase(collection.db);
 		lastCommand = MongoCommands.getCommand(`db.${collection.label}.find()`);
 		MongoCommands.executeCommand(lastCommand, connectedDb).then(result => util.showResult(result));
+		lastOpenedDocumentType = "Mongo";
 	});
 
 	initCommand(context, 'cosmosDB.launchMongoShell', () => launchMongoShell());
-	initAsyncCommand(context, 'cosmosDB.openDocDBCollection', async (collection: DocDBCollectionNode) => {
-		util.showResult(JSON.stringify(await collection.getDocuments(), null, 2));
+	initAsyncCommand(context, 'cosmosDB.openDocDBDocument', async (document: DocDBDocumentNode) => {
+		const masterKey = await document.coll.db.getPrimaryMasterKey();
+		const endpoint = await document.coll.db.getEndpoint();
+		lastOpenedDocument = document;
+		util.showResult(JSON.stringify(document.data, null, 2));
+		lastOpenedDocumentType = "DocDB";
```

Comment:
By design this flag has to modify if and only if `util.showResult` is called. Is there a clean way to automatically enforce this?

Reply:
This is fine for now